src/atom_finder/classifier/operator-precedence.clj

```diff
@@ -0,0 +1,131 @@
+(defmethod operator-group IASTBinaryExpression [node]
+  (if (assignment? node)
+    :assign
+
+    (let [operator-group-table
+               {IASTBinaryExpression/op_modulo :non-asso
+                IASTBinaryExpression/op_multiply :multiply
+                IASTBinaryExpression/op_divide :non-asso
+                IASTBinaryExpression/op_plus :add
+                IASTBinaryExpression/op_minus :non-asso
+                IASTBinaryExpression/op_shiftLeft :bitwise
+                IASTBinaryExpression/op_shiftRight :bitwise
+                IASTBinaryExpression/op_greaterThan :compare
+                IASTBinaryExpression/op_greaterEqual :compare 
+                IASTBinaryExpression/op_lessThan :compare
+                IASTBinaryExpression/op_lessEqual :compare
+                IASTBinaryExpression/op_equals :compare
+                IASTBinaryExpression/op_notequals :compare
+                IASTBinaryExpression/op_binaryAnd :bitwise
+                IASTBinaryExpression/op_binaryXor :bitwise
+                IASTBinaryExpression/op_binaryOr :bitwise
+                IASTBinaryExpression/op_logicalAnd :and
+                IASTBinaryExpression/op_logicalOr :or}]
+           (->> node .getOperator operator-group-table))))
+
+
+(defn specific-confusing-operator-combination?
+  "if the combination of groups of operators exists in this set, then the combination is confusing"
+  [combination]
+  (some #(= % combination) (map sort #{[:de_incr :pointer] [:multiply :add] [:arith_unary :add] [:arith_unary :multiply] [:and :add] [:and :multiply] [:and :arith_unary] [:or :add] [:or :multiply] [:or :arith_unary] [:or :and] [:not :add] [:not :multiply] [:not :arith_unary] [:not :and] [:not :or]
+;;[:compare :and] [:compare :or] // Underclassify
+[:compare :not] [:compare :compare] [:pointer :add] [:cond :arith_unary] [:cond :and] [:cond :or] [:cond :not] [:cond :compare] [:cond :cond] [:non-asso :add] [:non-asso :multiply] [:non-asso :arith_unary] [:non-asso :and] [:non-asso :or] [:non-asso :not] [:non-asso :non-asso] [:field_ref :pointer]})))
+
+(def always-confusing-operator-groups
+  "If any of these operator groups is used along with another operator, then it is confusing"
+  #{:bitwise :comma})
+
+(defn confusing-operator-combination?
+  "Is this combination of the operator groups confusing?"
+  [combination]
+  (or (specific-confusing-operator-combination? combination)
+      (some (partial contains? always-confusing-operator-groups) combination)))
+
+(defn operator-group-pair?
+  [node]
+  (and (operator-group node)
+       (some operator-group (children node))))
+
+(defn group-pair
+  "returns a collection of operator group pairs between the node and its children
+  , if a second parameter is passed, use that as the collection instead of the children"
+  ([node]
+   (->> node
+        children
+        (map operator-group)
+        (remove nil?)
+        (map (fn [child-group] (sort [(operator-group node) child-group])))))
```

Comment:
This is a very common pattern of have 2 function arities, both of which do the same thing, one using a predefined value. I usually define the unary version in terms of the binary:

    ([node] (group-pair node (children node)))